test/integration/jobset_controller_test.go

```diff
@@ -71,9 +70,8 @@ var _ = ginkgo.Describe("JobSet controller", func() {
 	// jobSetUpdate contains the mutations to perform on the jobset and the
 	// checks to perform afterwards.
 	type jobSetUpdate struct {
-		jobUpdateFn             func(*batchv1.JobList)
-		checkJobSetState        func(*jobset.JobSet)
-		expectedJobSetCondition jobset.JobSetConditionType
+		updateFn         func(*jobset.JobSet)
+		updateShouldFail bool
```

Comment:
I would flip this to updateShouldSucceed since currently all cases expect failure, and so we can remove all those `updateShouldFail: true` lines

Reply:
Makes sense, done

Reply:
Ah i misread and thought you were referring to jobset creation not jobset update, sorry about that. Updated.